### DISCUSSION TITLE
Predefined control input for a RC Fixed Wing UAV in JSBSim with Arduplane SITL

### DISCUSSION THREAD

**Discussion by rishi1906** [JSBSim-Team/jsbsim#1005] [General]
I am trying to validate the results of JSBSim with actual flight data for a fixed-wing UAV. 

I tried using tuned PID gains in SITL with JSBSim, but the trajectories seem to be way off when compared to actual flight.

I have attached comparison of both in a 3D plot.
![image](https://github.com/JSBSim-Team/jsbsim/assets/16606630/54f31160-6d5f-4b04-bb07-fba96550b96b)
![image](https://github.com/JSBSim-Team/jsbsim/assets/16606630/a28c9780-11a7-4e72-b207-8e1d6b9d515c)

Is there a way to force JSBSim to follow a set of predefined control (in this case control from actual flight) to see how it behaves with SITL. 

@seanmcleod could you suggest what shall I debug to understand the mismatch between trajectories?

**Comment by seanmcleod** [comment]
So just to confirm, is the JSBSim model supposed to be an accurate model of the aircraft you have actual flight data for?

**Reply by rishi1906** [reply]
Yes, that's exactly what I want to validate.

**Comment by seanmcleod** [comment]
So is suggest starting with much simpler tests to confirm your aerodynamic model.

For example take a roughly straight and level segment from the recorded flight data and confirm that with the same starting conditions at the start of the segment, including the control positions that the next say 60s of the segment matches fairly closely with your JSBSim output.

Then look for segments where they're is a small step change in a control like the elevator and see if your jsbsim model matches the dynamics etc.

**Comment by rishi1906** [comment]
"confirm that with the same starting conditions at the start of the segment, including the control positions"
How do I set these conditions and controls for JSBSim with Ardupilot Software in the Loop Simulation (SITL)?

**Comment by seanmcleod** [comment]
For this sort of initial testing I would keep Ardupilot SITL out of the loop.

Simply do it via plain JSBSim.
